app/src/main/java/com/arygm/quickfix/ui/search/ExpandableCategoryItem.kt

```diff
@@ -0,0 +1,135 @@
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.arygm.quickfix.model.categories.WorkerCategory
+import com.arygm.quickfix.model.categories.painting.PaintingCategory
+import com.arygm.quickfix.ressources.C
+import com.arygm.quickfix.ui.theme.poppinsFontFamily
+
+@Composable
+fun ExpandableCategoryItem(
+    item: WorkerCategory,
+    isExpanded: Boolean,
+    backgroundColor: Color = colorScheme.surface,
+    onExpandedChange: (Boolean) -> Unit,
+) {
+  val subCategories = remember { PaintingCategory.entries.toList() }
+  val interactionSource = remember { MutableInteractionSource() }
+  val rotationAngle by animateFloatAsState(targetValue = if (isExpanded) 180f else 0f, label = "")
+
+  Column(
+      modifier =
+          Modifier.fillMaxWidth()
+              .shadow(5.dp, shape = RoundedCornerShape(8.dp), clip = false)
+              .background(color = colorScheme.surface, shape = RoundedCornerShape(12.dp))
+              .clickable(interactionSource = interactionSource, indication = null) {
+                onExpandedChange(!isExpanded)
+              }
+              .semantics { testTag = C.Tag.expandableCategoryItem }) {
+        Row(
+            modifier =
+                Modifier.padding(start = 12.dp, top = 8.dp, bottom = 8.dp)
+                    .background(backgroundColor),
+            verticalAlignment = Alignment.CenterVertically) {
+              // Icon
+              Icon(
+                  imageVector = item.icon,
+                  contentDescription = null,
+                  tint = colorScheme.primary,
+                  modifier = Modifier.testTag("categoryIcon"))
+
+              Spacer(modifier = Modifier.width(10.dp))
+
+              // Text Column
+              Column(modifier = Modifier.weight(7f)) {
+                Text(
+                    text = item.displayName,
+                    color = colorScheme.onBackground,
+                    fontWeight = FontWeight.SemiBold,
+                    fontFamily = poppinsFontFamily,
+                    fontSize = 16.sp)
+                Text(
+                    text = item.description,
+                    color = colorScheme.onSecondary,
+                    fontWeight = FontWeight.Medium,
+                    fontFamily = poppinsFontFamily,
+                    fontSize = 11.sp,
+                    lineHeight = 16.sp)
+              }
+              Icon(
+                  imageVector = Icons.Filled.KeyboardArrowDown,
+                  contentDescription = if (isExpanded) "Collapse" else "Expand",
+                  modifier = Modifier.graphicsLayer(rotationZ = rotationAngle).weight(1f))
+            }
+        AnimatedVisibility(
+            visible = isExpanded,
+            enter = expandVertically() + fadeIn(),
+            exit = shrinkVertically() + fadeOut()) {
```

Comment:
The animations are fluid and the expandable item looks really good and matches the Figma, excellent work!